src/lib/snap_to.js

```diff
@@ -0,0 +1,120 @@
+const StringSet = require('../lib/string_set');
+const coordEach = require('@turf/meta').coordEach;
+const point = require('turf-point');
+const pointOnLine  = require('@turf/point-on-line');
+const distance = require('@turf/distance');
```

Comment:
https://github.com/mapbox/cheap-ruler is ~31x faster than @turf/distance, and over small distances (which are what's a play here) the margin of error is negligible. Perhaps you could use cheap-ruler instead?